pkg/scheduler/algorithm/predicates/metadata_test.go

```diff
@@ -507,3 +507,235 @@ func TestPredicateMetadata_ShallowCopy(t *testing.T) {
 		t.Errorf("Copy is not equal to source!")
 	}
 }
+
+// TestGetPodsMatchingAffinity_AntiAffinity tests againt method getPodsMatchingAffinity
+// on Anti Affinity cases
+func TestGetPodsMatchingAffinity_AntiAffinity(t *testing.T) {
```

Comment:
Maybe I am verbose, but I am thinking if you can add a test case to verify this bug is fixed in `predicates.go` as well?

Because we still have many logics there after we got the `antiAffinityPods` from metadata, so it would be great if we can ensure this change does work and will not break anything in `predicates.go`.

Sounds reasonable?

Reply:
@resouer Thanks for the detailed review!

Sure, it's totally reasonable. Do you think is it good enough to add additional UTs for `predicates.go#satisfiesPodsAffinityAntiAffinity()`? (step 2 of symmetry check this PR tries to fix)

Reply:
Yeah, should be.

Or, I think you can just add one more test case to verify this in `TestInterPodAffinity`? which seems quicker?

Reply:
Added one testcase - which only passed with this PR. PTAL.